ui/components/app/assets/asset-list/asset-list-control-bar/asset-list-control-bar.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  Box,
+  ButtonBase,
+  ButtonBaseSize,
+  IconName,
+  Popover,
+  PopoverPosition,
+} from '../../../../component-library';
+import { TokenWithBalance } from '../asset-list';
+import {
+  getPreferences,
+  getSelectedAccount,
+  getShouldHideZeroBalanceTokens,
+} from '../../../../../selectors';
+import SortControl from '../sort-control';
+import {
+  BackgroundColor,
+  BorderColor,
+  BorderStyle,
+  Display,
+  JustifyContent,
+  TextColor,
+} from '../../../../../helpers/constants/design-system';
+import { sortAssets } from '../../util/sort';
+import { useNativeTokenBalance } from '../native-token/use-native-token-balance';
+import ImportControl from '../import-control';
+import { useAccountTotalFiatBalance } from '../../../../../hooks/useAccountTotalFiatBalance';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+
+type AssetListControlBarProps = {
+  setTokenList: (arg: TokenWithBalance[]) => void;
+  showTokensLinks?: boolean;
+};
+
+const AssetListControlBar = ({
+  setTokenList,
+  showTokensLinks,
+}: AssetListControlBarProps) => {
+  const t = useI18nContext();
+  const controlBarRef = useRef<HTMLDivElement>(null); // Create a ref
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  const { useNativeCurrencyAsPrimaryCurrency, tokenSortConfig } =
```

Comment:
Note that `useNativeCurrencyAsPrimaryCurrency` will be removed in https://github.com/MetaMask/metamask-extension/pull/27097

Reply:
Yep, gonna have some conflicts here that Sahar and I will need to work through. IMO I think we should probably merge her feature before this one.